app/models/mentor_to_mentee_matcher.rb

```diff
@@ -62,6 +64,36 @@ def run_algorithm(mentors, mentees, with_time_zone)
     end
   end
 
+  def has_programming_language(mentee, language_slug)
+    mentee.programming_language.slug == language_slug || mentee.other_programming_languages.include?(language_slug)
+  end
+
+  def timezone_within_range(mentor, mentee)
+    (timezone_difference(mentor.time_zone, mentee.time_zone) <= 2)
+  end
+
+  def matches_experience(mentor, mentee, language_slug)
+    mentee_prog_level = programming_level(mentee.programming_experience_level[language_slug])
+    mentor_prog_level = programming_level(mentor.programming_experience_level[language_slug])
+
+    mentor_accepts_beginners(mentor, mentee_prog_level) && mentee_prog_level <= mentor_prog_level
+  end
+
+  def mentor_accepts_beginners(mentor, mentee_prog_level)
+    (mentor.like_mentoring_beginner || (!mentor.like_mentoring_beginner && mentee_prog_level != 1))
```

Comment:
I have the following questions
1. So when a mentor like mentoring a beginner, we don't check if it's a beginner or advanced. 
3. So what happen if the mentee_prog_level is advanced and mentor is beginner ? This case is not covered.

Reply:
I'll start by answering question 2 first. 

> 2. So what happen if the mentee_prog_level is advanced and mentor is beginner ? This case is not covered.

If you check the line 79, you see the condition `mentee_prog_level <= mentor_prog_level` which checks the compatibility of programming levels. So it will only matches if the mentor level is equal or greather than the mentee. If the mentee is advanced and mentor is beginner, the condition fails and they will not match.

> 1. So when a mentor like mentoring a beginner, we don't check if it's a beginner or advanced.

We don't have to check whether the mentor is beginner or not because if the mentor like is a beginner, it wil only matches with a beginner mentee due to the condition in line 79.

Let me know if you need further clarification.

Reply:
sure sound good. I missed this check 
mentee_prog_level <= mentor_prog_level
Can you add some comment in the code as well ? for same.

Reply:
Done.